vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/wicket/model/SolrDocumentModel.java

```diff
@@ -34,14 +39,17 @@
  */
 public class SolrDocumentModel extends LoadableDetachableModel<SolrDocument> {
 
+    
     private final IModel<String> docId;
+    
+
 
-    public SolrDocumentModel(SolrDocument document) {
+    public SolrDocumentModel(SolrDocument document, FieldNameService fieldNameService) {
```

Comment:
Here I would suggest getting the `FieldNameService` instance by calling `VloWicketApplication.get().getFieldNameService()` where needed, which takes away the need to pass around the service in a couple of places. The models are not used outside the context . If this causes trouble in the tests (as there will not be a `VloWicketApplication` context), you could make an overrideable method for getting a field name service instance which can be overridden in the test (where a mock or actually available instance is returned instead).